src/Table/Table.tsx

```diff
@@ -227,6 +227,118 @@ const ActionColumn: React.FunctionComponent<ActionColumnProps> = (props: ActionC
     );
 };
 
+interface RowUIProps {
+    actionLinks?: Array<ActionLinkItem>;
+    columns: Array<TableHeader>;
+    onActionDropped: (event: React.MouseEvent<HTMLDivElement, MouseEvent>, row: TableRow, rowIndex?: number) => void;
+    onItemSelected?: (e: React.ChangeEvent<HTMLInputElement>, row: TableRow, type: RowTypes, rowIndex?: number) => void;
+    onRowExpanded?: (e: React.MouseEvent<HTMLDivElement, MouseEvent>, row: TableRow) => void;
+    onSubRowExpanded?: (e: React.MouseEvent<HTMLDivElement, MouseEvent>, row: TableRow, rowIndex: number) => void;
+    parentRowIndex?: number;
+    parentRowIsExpanded?: boolean;
+    primaryActionButton?: PrimaryActionButton;
+    row: TableRow;
+    rowsAreCollapsable: boolean;
+    tableRef: React.RefObject<HTMLTableElement>;
+    type: RowTypes;
+    useRowCollapse: boolean;
+    useRowSelection: boolean;
+    useShowActionColumn: boolean;
+}
+
+const RowUI: React.FunctionComponent<RowUIProps> = (props: RowUIProps) => {
```

Comment:
Could you name this `TableRow`? I think it's more descriptive.

Reply:
We have an interface TableRow already, which is an interface for tableRow. I call this UI because it's UI. Also you notice we have another UI tableUI, which is the main UI for the table. So basically UI here is my way of categorizing them.